### PR TITLE
Add support for ListRulesMachines Endpoint

### PR DESCRIPTION
**Pull request by stack72**
@cheapRoc

### CODE REVIEW

File: network/firewall_test.go [hunk 1]
```diff
@@ -0,0 +1,196 @@
...
+
+func listRuleMachinesSuccess(req *http.Request) (*http.Response, error) {
+	header := http.Header{}
+	header.Add("Content-Type", "application/json")
+
+	body := strings.NewReader(`[
+	{
+    "id": "b6979942-7d5d-4fe6-a2ec-b812e950625a",
+    "name": "test",
+    "type": "smartmachine",
```
Comment (cheapRoc) [inline network/firewall_test.go]: Is there anything different between a Docker container or native machine? Are we testing that?

File: network/firewall.go [hunk 2]
```diff
@@ -248,3 +250,52 @@ func (c *FirewallClient) ListMachineRules(ctx context.Context, input *ListMachin
...
+	Brand           string                 `json:"brand"`
+	State           string                 `json:"state"`
+	Image           string                 `json:"image"`
+	Memory          int                    `json:"memory"`
+	Disk            int                    `json:"disk"`
+	Metadata        map[string]string      `json:"metadata"`
+	Tags            map[string]interface{} `json:"tags"`
+	Created         time.Time              `json:"created"`
+	Updated         time.Time              `json:"updated"`
+	Docker          bool                   `json:"docker"`
```
Comment (cheapRoc) [inline network/firewall.go]: Minor nit but I don't see this exercised through the new API's unit tests. See note below.

File: network/firewall_test.go (hunk 1 above)
Comment (stack72) [reply network/firewall_test.go]: We are not testing that - the API itself should take care of that - there are no docs about what is supported here. We will just return a list of machines as expected

Comment (cheapRoc) [reply network/firewall_test.go]: 👍 Pretty much all I wanted to know.

**Pull request merged**